src/main/java/de/themorpheus/edu/taskservice/controller/solution/SolutionController.java

```diff
@@ -35,14 +47,19 @@
 		return ControllerResult.of(solutionModel);
 	}
 
-	public ControllerResult<SolutionModel> deleteSolution(SolutionModel solutionModel) {
-		this.solutionRepository.delete(solutionModel);
+	public ControllerResult<SolutionModel> deleteSolution(int taskId) {
+		ControllerResult<SolutionModel> solutionModel = this.getSolution(taskId);
+		if (solutionModel.isResultNotPresent()) return ControllerResult.ret(solutionModel);
+
+		this.solutionRepository.delete(solutionModel.getResult());
+
+		for (SolutionInterface solutionInterface : solutionInterfaces) solutionInterface.deleteAll(solutionModel.getResult().getTaskId().getTaskId());
```

Comment:
solutionInterface -> solutionController
because it's not an interface, it's a controller instance implementing "SolutionInterface"